### PR TITLE
528 rewrite wrong input

### PR DESCRIPTION
**Pull request by index-git**
Part of issue #528

- [x] Tests
- ~~Layman Test Client (including docker image at docker hub)~~
- ~~Changelog~~
- ~~Documentation~~

### CODE REVIEW

File: tests/dynamic_data/publications/wrong_input/wrong_input_test.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        KEY_ACTION_PARAMS: {
+            'file_paths': [
+                'tmp/naturalearth/110m/cultural/ne_110m_admin_0_boundary_lines_land.cpg',
+                'tmp/naturalearth/110m/cultural/ne_110m_admin_0_boundary_lines_land.README.html',
+                'tmp/naturalearth/110m/cultural/ne_110m_admin_0_boundary_lines_land.shp',
+                'tmp/naturalearth/110m/cultural/ne_110m_admin_0_boundary_lines_land.shx',
+                'tmp/naturalearth/110m/cultural/ne_110m_admin_0_boundary_lines_land.VERSION.txt',
+            ],
+        },
+        consts.KEY_EXCEPTION: LaymanError,
```
Comment (jirik) [inline tests/dynamic_data/publications/wrong_input/wrong_input_test.py:47]: Use local variable/enum value instead of imported one

**Review by jirik — CHANGES_REQUESTED** [review]

File: tests/dynamic_data/publications/wrong_input/wrong_input_test.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        exp_exception = copy.deepcopy(params[KEY_EXPECTED_EXCEPTION])
+        is_sync = exp_exception.pop('sync')
+        exception = pytest.raises(params[consts.KEY_EXCEPTION]) if is_sync else does_not_raise()
+        with exception as exception_info:
+            response = rest_method(publication, args=rest_args)
+        if is_sync:
+            processing.exception.response_exception(expected=exp_exception, thrown=exception_info)
+        #     Po patchi je validní publikace
+        else:
+            processing.response.valid_post(workspace=publication.workspace,
```
[the remark this answers is not in the document]
Comment (index-git) [reply tests/dynamic_data/publications/wrong_input/wrong_input_test.py:139]: We tested only common keys and process_client returns the first item from the response array.

File: tests/dynamic_data/publications/wrong_input/wrong_input_test.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+from layman import LaymanError
+from test_tools import process_client
+from tests import EnumTestTypes, Publication
+from tests.asserts import processing
+from tests.asserts.final import publication as publication_asserts
+from tests.asserts.final.publication import util as assert_utils
+from tests.dynamic_data import base_test
+
+
+class KEYS(Enum):
```
Comment (jirik) [inline tests/dynamic_data/publications/wrong_input/wrong_input_test.py:17]: KEYS -> Keys

Comment (jirik) [reply tests/dynamic_data/publications/wrong_input/wrong_input_test.py:17]: Add unique constraint

**Review by jirik — CHANGES_REQUESTED** [review]

**Review by jirik — APPROVED** [review]

**Pull request merged**